rfcs/none-and-unknown-types.md

```diff
@@ -0,0 +1,99 @@
+# Add never and unknown types
+
+## Summary
+
+Add `unknown` and `never` types that are inhabited by everything and nothing respectively.
+
+## Motivation
+
+There are lots of cases in local type inference, semantic subtyping,
+and type normalization, where it would be useful to have top and
```

Comment:
One more motivation! It would also be useful to know when a branch can actually ever be taken.

Here we first narrow `x` to be number, which we then narrow it not to be a number, so `typeof(x) & number & ~number` which is normalized into `never`. Control flow analysis could run through this branch to look for errors but never apply side effects to the rest of the scope!
```lua
if type(x) == "number" and type(x) ~= "number" then
  ...
end
```